internal/config/pipeline_test.go

```diff
@@ -128,6 +128,71 @@ func TestPipeline(t *testing.T) {
 	}
 }
 
+// This test is a little weird. We parse a pipeline, but only to use its config
+// to get to the HCL Eval context for loading all pipelines and generating the
+// Proto slice. This test intends to show that if you eval an HCL context,
+// the PipelineProtos func will load those and convert them to Proto Pipelines.
+// So there's probably a better way to get a pipeline config struct to access
+// PipelineProtos()?
+func TestPipelineProtos(t *testing.T) {
```

Comment:
This test makes me think it might make more sense to have the pipeline `ConfigSync` take an `*App` pointer receiver instead? I wasn't able to get at `PipelineProtos()` when I did that though, so maybe this is ok for now. It might just depend on what the runner has access to when it attempts to call `ConfigSync`?

Reply:
The way I'd approach this instead is probably to pass `*Config` into the `Func` field rather than `Pipeline`. Then you just call `c.PipelineProtos()` and assert on that. The rest looks fine to me...